src/scanner.c

```diff
@@ -258,6 +259,38 @@ bool is_valid_string_delim(char c) {
   return c == '\'' || c == '"';
 }
 
+// Symbols allowed as pattern delimiter
+bool is_valid_syn_pattern_delimiter(char c) {
+  static char valids[] = {
+    '-',
+    '`',
+    '~',
+    '!',
+    '@',
+    '#',
+    '$',
+    '^',
+    '&',
+    '*',
+    '_',
+    '=',
+    '+',
+    ';',
+    ':',
+    ',',
+    '/',
+    '?',
+    '\'',
+    '"',
+  };
+
+  for (int i = 0; i < sizeof(valids) / sizeof(valids[0]); ++i)
```

Comment:
I don't really like those, could you please either make the `valids` a string, and check for `'\0'` or have the size of the array as a preprocessor variable ?